cmake/Fortran.cmake

```diff
@@ -18,34 +18,80 @@ if(APPLE)
     # master of gcc has a fix via a new `-static-libquadmath` but it's not even in gcc 12.2.0. So instead we have to do all kinds of shenanigans
 
     enable_language(Fortran)  # Needed to populate ${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES}
+
+    include(CheckFortranCompilerFlag)
+    check_fortran_compiler_flag(-static-libgfortran HAS_STATIC_LIBGFORTRAN)
+    check_fortran_compiler_flag(-static-libgcc HAS_STATIC_LIBGCC)
+    check_fortran_compiler_flag(-static-libquadmath HAS_STATIC_LIBQUADMATH)
+    message(DEBUG "HAS_STATIC_LIBGFORTRAN=${HAS_STATIC_LIBGFORTRAN}, HAS_STATIC_LIBGCC=${HAS_STATIC_LIBGCC}, HAS_STATIC_LIBQUADMATH=${HAS_STATIC_LIBQUADMATH}")
     # Debug:
     # set(Fortran_VERBOSE "-v")
-    # Find the static libquadmath maually
-    find_library(static-libquadmath NAMES quadmath.a libquadmath.a REQUIRED PATHS ${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES})
-    message(STATUS "Found static-libquadmath at ${static-libquadmath} by searching in CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES=${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES}")
+
     target_compile_options(fortran_project_options INTERFACE
       ${Fortran_VERBOSE}
     )
-    target_link_options(fortran_project_options INTERFACE
-      ${Fortran_VERBOSE}
-      -nodefaultlibs
-      -static-libgfortran
-      -lgfortran
-      -static-libgcc
-      -lgcc
-      -lemutls_w
-      ${static-libquadmath}
-      -lSystem
-    )
 
-    message(DEBUG "CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES-${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES}")
-    #    => CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES-gfortran;emutls_w;gcc;quadmath;emutls_w;gcc;gcc
-    list(REMOVE_DUPLICATES CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES)
-    list(TRANSFORM CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES REPLACE "quadmath" " ${static-libquadmath}")
-    message(DEBUG "CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES-${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES}")
+    if (NOT HAS_STATIC_LIBGFORTRAN OR NOT HAS_STATIC_LIBGCC)
+      message(FATAL_ERROR "Please check your compiler. gfortran supports -static-libgcc and -static-libgfortran since at least 2007, but HAS_STATIC_LIBGFORTRAN=${HAS_STATIC_LIBGFORTRAN}, HAS_STATIC_LIBGCC=${HAS_STATIC_LIBGCC}")
```

Comment:
I think this is quite reasonable.  And kind wording "Please check your compiler..."